File: internal/model/album.go
package model

import "time"

// Album represents a music album in the library.
type Album struct {
	ID              string    `json:"id"`
	ArtistID        string    `json:"artistId"`
	Name            string    `json:"name"`
	Year            int       `json:"year,omitempty"`
	Genre           string    `json:"genre,omitempty"`
	CoverArtPath    string    `json:"-"`
	TrackCount      int       `json:"trackCount"`
	DurationSeconds int       `json:"durationSeconds"`
	CreatedAt       time.Time `json:"createdAt"`
}

File: ios/Sonus/Models/Album.swift
import Foundation

/// A music album in the library.
struct Album: Codable, Identifiable, Hashable, Sendable {
    let id: String
    let artistId: String
    let name: String
    let year: Int?
    let genre: String?
    let trackCount: Int
    let durationSeconds: Int
    let createdAt: Date
}
